js/script.js

```diff
@@ -0,0 +1,153 @@
+const newPlayer = (name, symbol, turn) => {
+  const playerName = name;
+  const token = symbol;
+  const playerTurn = turn;
+  return { playerName, token, playerTurn };
+};
+
+const gameBoard = () => {
+  let board;
+
+  const blockCells = () => {
+    const cells = document.querySelectorAll('.cell');
+    [...cells].forEach(cell => { cell.style.pointerEvents = 'none'; });
+  };
+
+  const unblockCells = () => {
+    const cells = document.querySelectorAll('.cell');
+    [...cells].forEach(cell => { cell.style.pointerEvents = 'auto'; });
+  };
+
+  const init = () => {
+    board = [null, null, null, null, null, null, null, null, null];
+    for (let i = 0; i < 9; i += 1) {
+      document.getElementById('cell' + i).innerHTML = ''; // eslint-disable-line prefer-template
+    }
+    document.getElementById('winner-text').innerHTML = '';
+    unblockCells();
+  };
+
+  const setCell = (index, symbol) => {
+    board[index] = symbol;
+  };
+
+  const checkWinCondition = () => {
+    let isWin = false;
+    if (board[0] != null && board[0] === board[1] && board[1] === board[2]) isWin = true;
+    if (board[3] != null && board[3] === board[4] && board[4] === board[5]) isWin = true;
+    if (board[6] != null && board[6] === board[7] && board[7] === board[8]) isWin = true;
+    if (board[0] != null && board[0] === board[3] && board[3] === board[6]) isWin = true;
+    if (board[1] != null && board[1] === board[4] && board[4] === board[7]) isWin = true;
+    if (board[2] != null && board[2] === board[5] && board[5] === board[8]) isWin = true;
+    if (board[0] != null && board[0] === board[4] && board[4] === board[8]) isWin = true;
+    if (board[6] != null && board[6] === board[4] && board[4] === board[2]) isWin = true;
+    return isWin;
+  };
+
+  const checkDrawCondition = () => {
+    let isDraw = true;
+    if (board.includes(null)) isDraw = false;
+    return isDraw;
+  };
+
+  return {
+    init,
+    setCell,
+    checkWinCondition,
+    checkDrawCondition,
+    blockCells,
+    unblockCells,
+  };
+};
+
+const gameLogic = () => {
+  let player1;
+  let player2;
+  const tileset = [];
+  const newGameBoard = gameBoard();
+
+  const setTile = () => {
+    for (let i = 0; i < 9; i += 1) {
+      tileset[i] = document.getElementById('cell' + i); // eslint-disable-line prefer-template
+    }
+  };
+
+  const init = () => {
+    newGameBoard.init();
+    player1 = newPlayer(document.getElementById('player01Name').value, 'X', true);
+    player2 = newPlayer(document.getElementById('player02Name').value, 'O', false);
+    document.getElementById('info').innerText = `${player1.playerName} 's move`;
+    setTile();
```

Comment:
Why not try to enforce the user experience by making sure that user files in proper user names here.

![Screen Shot 2020-05-20 at 11 09 30 AM](https://user-images.githubusercontent.com/40712459/82434004-2ed38580-9a8a-11ea-98a6-43487d9f8466.png)

If i validate this form without entering any username, it will work. You can also consider default names (like `player1`, `player2`), in case user doesn't fill anything